### DISCUSSION TITLE
Bug y una posible mejora

### DISCUSSION THREAD

**Discussion by Moiren1** [elpatronaco/pccomponentes-buy-bot#4] [General]
Buenas Pau, ante todo agradecerte el programa que has creado, quería comentarte unas cosas que he visto: han quitado el "Avisame" del stock, con lo cual el bot se cree que esta en stock.
por otro lado al poner 3-4 tareas para que busque, no siempre refresca las pantallas, se te ha ocurrido la posibilidad de poner poner varios enlaces y cada refresco sea en una pagina solo alternando? 

un saludo y muchas gracias por todo,

**Comment by elpatronaco** [comment]
Hola Moiren, parece que han quitado stock de casi todo 😶 que rayada. Las reacondicionadas sin stock siguen teniendo el botón con id 'notify-me', por lo que sigue funcionando según lo que he probado, supongo que te refieres a eso. 
Sobre lo otro, si que lo tengo pensado, y más ahora sabiendo que el mercado de las gráficas está en la puta mierda, peor que nunca. Pero no con selenium, ya que para estas tareas en las que se necesita cada milisegundo es muy muy lento. Quiero implementar un crawler con un paquete como [este](https://www.npmjs.com/package/crawler) el cual lance el bot de autocompra cuando detecte stock de alguna de las gráficas que le pases. También tengo pensado cambiar de Selenium a Puppeteer, con lo que ganaríamos unos segundos más yo creo. Gracias por los comentarios eh, ahora voy muy corto de tiempo con todo y dedicarle un poco más a esto es difícil, pero intentaré hacer lo que pueda. De nuevo gracias a ti!

**Comment by elpatronaco** [comment]
De momento las tareas que tengo son
- Refactorizar a Puppeteer
- Implementar crawler para recorrer el dom más rápidamente
- Implementar avisos por Telegram
Las iré haciendo en orden puesto que cambiar a puppeteer no me debería llevar más de 20 minutos, y es un cambio que creo que se va a notar

**Comment by Moiren1** [comment]
Muchas gracias por las respuestas y por tu trabajo, yo he tenido que comprar una 3090 mientras espero impaciente el poder pillar una 3080, con tu bot esperaba poder conseguirlo.
ahora revisare lo que hay en la web y aunque sea pondré el bot a buscar reacondicionadas, pero si las nuevas desaparecen, las reacondicionadas están mas solicitadas que los bitcoins XD.
Cualquier ayuda que pueda brindarte dímelo. 
un saludo

**Comment by elpatronaco** [comment]
😂 ahora queda toda la roña. Yo si fuera tú tiraba con la 3090, sé que te habrá costado medio riñón pero te ahorras el dolor de cabeza de buscar una 3080

**Comment by elpatronaco** [comment]
He hecho un arreglo un poco rápido del bug que me dijiste ayer

**Comment by Moiren1** [comment]
voy a probar, muchas gracias

**Comment by Moiren1** [comment]
> 😂 ahora queda toda la roña. Yo si fuera tú tiraba con la 3090, sé que te habrá costado medio riñón pero te ahorras el dolor de cabeza de buscar una 3080

el día 1 dono el riñon :'P

**Comment by Moiren1** [comment]
ya lo tengo en marcha, en principio funciona bien, falta ver cuando aparezca en stock que ocurre^^, por otro lado has añadido la función del telegram, me da error en la línea phone,
lo he comentado y así tira, igualmente falta ver cuando quieras mandar un mensaje que ocurre al tener yo comentada esa parte.
un saludo y gracias.

**Comment by elpatronaco** [comment]
Fuck, me pasa siempre lo mismo, como en el index.ts tengo mi cuenta y mi tarjeta nunca hago commit de eso y siempre da fallos. Comenta todo lo que veas de telegram y reza porque no falle, cuando llegue a casa lo quito hasta que no esté hecho. Vaya mierda d developer estoy hecho

**Comment by joanroig** [comment]
> Fuck, me pasa siempre lo mismo, como en el index.ts tengo mi cuenta y mi tarjeta nunca hago commit de eso y siempre da fallos. Comenta todo lo que veas de telegram y reza porque no falle, cuando llegue a casa lo quito hasta que no esté hecho. Vaya mierda d developer estoy hecho

Hola, quizás puedes usar un fichero JSON de configuración como éste con la info de la cuenta de compra para dejar el index.ts limpio: https://github.com/joanroig/pccomponentes-outlet-bot/blob/master/config.json

**Comment by elpatronaco** [comment]
@joanroig estoy refactorizando todo el bot para implementar el crawler, cuando acabe pondré algo así, mersi. Pero de momento quería dejar la opción de instanciar el bot las veces que quieras para la gente que quiera buscar más gráficas

**Comment by Rogue29x** [comment]
> ya lo tengo en marcha, en principio funciona bien, falta ver cuando aparezca en stock que ocurre^^, por otro lado has añadido la función del telegram, me da error en la línea phone,
> lo he comentado y así tira, igualmente falta ver cuando quieras mandar un mensaje que ocurre al tener yo comentada esa parte.
> un saludo y gracias.

Me puedes echar una mano si no es mucha molestia en ponerlo en marcha? no tengo narices a conseguirlo, lo he intentado todo pero me tira un error. Ya sea por telegram, dc, donde puedas y quieres claro. Te lo agradeceria, se me ha jodido la grafica y quiero comprar algo decente

**Comment by Moiren1** [comment]
> > ya lo tengo en marcha, en principio funciona bien, falta ver cuando aparezca en stock que ocurre^^, por otro lado has añadido la función del telegram, me da error en la línea phone,
> > lo he comentado y así tira, igualmente falta ver cuando quieras mandar un mensaje que ocurre al tener yo comentada esa parte.
> > un saludo y gracias.
> 
> Me puedes echar una mano si no es mucha molestia en ponerlo en marcha? no tengo narices a conseguirlo, lo he intentado todo pero me tira un error. Ya sea por telegram, dc, donde puedas y quieres claro. Te lo agradeceria, se me ha jodido la grafica y quiero comprar algo decente

pásame un pantallazo del error haber si puedo ayudarte en algo.

saludos

**Comment by Rogue29x** [comment]
Aquí te dejo el pantallazo

https://gyazo.com/d48d5b185cd56b29ec66fdbf51fe9467

**Comment by Moiren1** [comment]
copiame todo el texto del index.ts y borra los datos de tu cuenta y el telf

**Comment by elpatronaco** [comment]
@Rogue29x veo que te falta una coma por ahí. Descárgate el proyecto de nuevo y edita los textos dentro de comillas ' ' y los números sin tocar nada más

**Comment by Moiren1** [comment]
> @Rogue29x veo que te falta una coma por ahí. Descárgate el proyecto de nuevo y edita los textos dentro de comillas ' ' y los números sin tocar nada más

o eso XD te iba a repasar yo eso.

**Comment by Rogue29x** [comment]
Ya me funciona, lo descargué de nuevo y listo. La cuestión es si comprará la gpu, solo están las reacondicionadas

**Comment by elpatronaco** [comment]
Ayer salieron bastantes, yo guardé una 3060 Zotac 🤮 que por mala que sea ahí está, de momento el pescado está vendido pero dentro de unos días saldrán más. He abierto la pestaña de discusiones donde se puede hablar de lo q sea, y así mantenemos esto solo para bugs/pull requests/etc
